core/lib/multivm/src/versions/testonly/mod.rs

```diff
@@ -0,0 +1,213 @@
+    )
+}
+
+fn read_complex_upgrade() -> Vec<u8> {
+    read_bytecode("etc/contracts-test-data/artifacts-zk/contracts/complex-upgrade/complex-upgrade.sol/ComplexUpgrade.json")
+}
+
+fn read_precompiles_contract() -> Vec<u8> {
+    read_bytecode(
+        "etc/contracts-test-data/artifacts-zk/contracts/precompiles/precompiles.sol/Precompiles.json",
+    )
+}
+
+fn load_precompiles_contract() -> Contract {
+    load_contract(
+        "etc/contracts-test-data/artifacts-zk/contracts/precompiles/precompiles.sol/Precompiles.json",
+    )
+}
+
+fn read_proxy_counter_contract() -> (Vec<u8>, Contract) {
+    const PATH: &str = "etc/contracts-test-data/artifacts-zk/contracts/counter/proxy_counter.sol/ProxyCounter.json";
+    (read_bytecode(PATH), load_contract(PATH))
+}
+
+fn read_nonce_holder_tester() -> Vec<u8> {
+    read_bytecode("etc/contracts-test-data/artifacts-zk/contracts/custom-account/nonce-holder-test.sol/NonceHolderTest.json")
+}
+
+fn read_expensive_contract() -> (Vec<u8>, Contract) {
+    const PATH: &str =
+        "etc/contracts-test-data/artifacts-zk/contracts/expensive/expensive.sol/Expensive.json";
+    (read_bytecode(PATH), load_contract(PATH))
+}
+
+fn read_many_owners_custom_account_contract() -> (Vec<u8>, Contract) {
+    let path = "etc/contracts-test-data/artifacts-zk/contracts/custom-account/many-owners-custom-account.sol/ManyOwnersCustomAccount.json";
+    (read_bytecode(path), load_contract(path))
+}
+
+fn read_error_contract() -> Vec<u8> {
+    read_bytecode(
+        "etc/contracts-test-data/artifacts-zk/contracts/error/error.sol/SimpleRequire.json",
+    )
+}
+
+pub(crate) fn read_max_depth_contract() -> Vec<u8> {
+    read_zbin_bytecode(
+        "core/tests/ts-integration/contracts/zkasm/artifacts/deep_stak.zkasm/deep_stak.zkasm.zbin",
+    )
+}
+
+pub(crate) fn read_simple_transfer_contract() -> Vec<u8> {
+    read_bytecode(
+        "etc/contracts-test-data/artifacts-zk/contracts/simple-transfer/simple-transfer.sol/SimpleTransfer.json",
+    )
+}
+
+pub(crate) fn get_bootloader(test: &str) -> SystemContractCode {
+    let bootloader_code = read_bootloader_code(test);
+    let bootloader_hash = hash_bytecode(&bootloader_code);
+    SystemContractCode {
+        code: bytes_to_be_words(bootloader_code),
+        hash: bootloader_hash,
+    }
+}
+
+pub(super) fn default_system_env() -> SystemEnv {
+    SystemEnv {
+        zk_porter_available: false,
+        version: ProtocolVersionId::latest(),
+        base_system_smart_contracts: BaseSystemContracts::playground(),
+        bootloader_gas_limit: BATCH_COMPUTATIONAL_GAS_LIMIT,
+        execution_mode: TxExecutionMode::VerifyExecute,
+        default_validation_computational_gas_limit: BATCH_COMPUTATIONAL_GAS_LIMIT,
+        chain_id: L2ChainId::from(270),
+    }
+}
+
+pub(super) fn default_l1_batch(number: L1BatchNumber) -> L1BatchEnv {
+    let timestamp = unix_timestamp_ms();
```

Comment:
Please remove all nondeterminism. It would makes debugging easier and is required for CallTracer tests.